### PR TITLE
The current draft is now archived prior to being bumped in the DATASTORE/archive folder

### PR DESCRIPTION
**Pull request by essbb**

### CODE REVIEW

File: job_executor/adapter/local_storage.py [hunk 1]
```diff
@@ -360,3 +361,27 @@ def delete_temporary_backup() -> Union[None, LocalStorageError]:
...
+    """
+
+    archive_dir = Path(f'{DATASTORE_DIR}/archive')
+
+    if not archive_dir.exists():
+        os.makedirs(archive_dir, exist_ok = False)
+
+    timestamp = datetime.now()
+
+    archived_metadata_path = Path(
```
Comment (DanielElisenberg) [inline job_executor/adapter/local_storage.py:379]: (I love the slash operator with Path)
```py
    archived_draft_version_path = (
        archive_dir / f'draft_version_{version}_{timestamp}.json'
    )
```

File: job_executor/adapter/local_storage.py [hunk 2]
```diff
@@ -360,3 +361,27 @@ def delete_temporary_backup() -> Union[None, LocalStorageError]:
...
+    """
+    Archives the current draft json
+    * dataset_name: str - name of dataset draft
+    * version: str - version of the archived draft
+    """
+
+    archive_dir = Path(f'{DATASTORE_DIR}/archive')
+
+    if not archive_dir.exists():
+        os.makedirs(archive_dir, exist_ok = False)
```
Comment (DanielElisenberg) [inline job_executor/adapter/local_storage.py:375]: 💯

File: tests/unit/model/test_datastore.py [hunk 3]
```diff
@@ -212,6 +214,7 @@ def test_bump_datastore_minor():
         'KJOENN': 'KJOENN__0_1.parquet',
         'SIVSTAND': 'SIVSTAND__0_1.parquet'
     }
+    assert len(_get_file_list_from_dir(Path(DATASTORE_ARCHIVE_DIR))) > 0
```
Comment (DanielElisenberg) [inline tests/unit/model/test_datastore.py:217]: Would be nice if this expected 1

File: tests/unit/model/test_datastore.py [hunk 4]
```diff
@@ -265,3 +268,18 @@ def test_bump_datastore_major():
...
```
Comment (DanielElisenberg) [inline tests/unit/model/test_datastore.py:271]: > Would be nice if this expected 1
And then this expects 2 to make sure the previous archived file is not deleted

File: job_executor/model/datastore.py [hunk 5]
```diff
@@ -199,6 +199,9 @@ def bump_version(self, bump_manifesto: DatastoreVersion, description: str):
             [] if self.metadata_all_latest is None
             else [ds for ds in self.metadata_all_latest]
         )
+
+        local_storage.archive_draft_version(new_version)
```
Comment (DanielElisenberg) [inline job_executor/model/datastore.py:203]: this needs to be moved above line 191, due to line 191 mutating the draft file we wish to archive:
```py
if not self.draft_version.validate_bump_manifesto(bump_manifesto):  # currently line 186
    raise VersioningException(
        'Invalid Bump: Changes were made to the datastore '
        'after bump was requested'
    )
local_storage.archive_draft_version(self.latest_version_number)  # here
release_updates, update_type = self.draft_version.release_pending() # currently line 191
```
We can use the "pre-bump" version number 👍🏻

Comment (essbb) [reply job_executor/model/datastore.py:203]: Ahh, good point

File: tests/unit/model/test_datastore.py (hunk 4 above)
Comment (essbb) [reply tests/unit/model/test_datastore.py:271]: Yessir!

File: tests/unit/model/test_datastore.py (hunk 3 above)
Comment (essbb) [reply tests/unit/model/test_datastore.py:217]: Yup

File: job_executor/adapter/local_storage.py (hunk 1 above)
Comment (essbb) [reply job_executor/adapter/local_storage.py:379]: Nice :)

**Review by DanielElisenberg — APPROVED** [review]
Looks great :100:

**Pull request merged**